allenact/algorithms/onpolicy_sync/engine.py

```diff
@@ -916,12 +915,38 @@ def act(self, rollouts: RolloutStorage, dist_wrapper_class: Optional[type] = Non
 
         return actions, actor_critic_output, memory, step_observation
 
+    def advantage_stats(
+        self, advantages: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        r"""Computes the mean and variances of advantages (possibly over multiple workers).
+        For multiple workers, this method is equivalent to first collecting all versions of
+        advantages and then computing the mean and variance locally over that
+        :param advantages: (*,) shaped tensors to compute mean and variance over.  Assumed
+                            to be solely the workers local copy of this tensor,
+                            the resultant mean and variance will be computed
+                            over _all_ workers version of this tensor in distributed training.
+        """
+        if self.is_distributed:
+            mean = advantages.mean()
+            dist.all_reduce(mean)
+            mean = mean / self.num_workers
+
+            var = (advantages - mean).pow(2).mean()
+            dist.all_reduce(var)
+            std = (var / self.num_workers).sqrt()
+        else:
+            mean, std = advantages.mean(), advantages.std()
```

Comment:
Since we are already sharing the total number of steps across all workers, I changed the code to be more robust to uneven rollout sizes and added the Bessel's correction to the stdev computation. This deviates from the implementation in Habitat, but I agree it makes more sense from the math point of view.